tests/unit-tests/frontend_wayland/test_g_desktop_file_cache.cpp

```diff
@@ -43,11 +44,12 @@ struct GDesktopFileCache : Test
     static void SetUpTestSuite()
     {
         // Establish the temporary directory
-        std::filesystem::path tmp_dir_path {std::filesystem::temp_directory_path() /= std::tmpnam(nullptr)};
-        std::filesystem::create_directories(tmp_dir_path);
-        std::string desktop_file_directory_name = tmp_dir_path;
-        setenv("XDG_DATA_DIRS", desktop_file_directory_name.c_str(), 1);
-        applications_dir = desktop_file_directory_name + "/applications";
+        char tmp_dir_path[] = "/tmp/mir-XXXXXX";
```

Comment:
Why are we dropping `filesystem::temp_directory_path()` in favor of hard-coding `/tmp/`?

Reply:
I dropped it as it would make it harder to make the array to pass to `mkdtemp`. In many other tests we have hardcoded `/tmp` and this seems reasonable (extremely unlikely to need to support another directory for temporary files).

Reply:
Perhaps it might help to rely on `std::string` rather than C char arrays? Something like:

```c++
tmp_dir = (std::filesystem::temp_directory_path() /= "mir-XXXXXX").string();
ASSERT_THAT(mkdtemp(tmp_dir.data()), NotNull());
```

The writable char array returned by `data()` is guaranteed to be null terminated, so is safe to pass to mkdtemp here. We can't do this with a `std::filesystem::path`, since it doesn't give access to the path's backing array.